modules/images/webp-uploads/load.php

```diff
@@ -54,51 +55,40 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 		wp_update_attachment_metadata( $attachment_id, $metadata );
 	}
 
-	$allowed_mimes      = array_flip( wp_get_mime_types() );
+	$original_size_data = array(
+		'width'  => isset( $metadata['width'] ) ? (int) $metadata['width'] : 0,
+		'height' => isset( $metadata['height'] ) ? (int) $metadata['height'] : 0,
+		'crop'   => false,
+	);
+
 	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
-	$original_extension = pathinfo( $file, PATHINFO_EXTENSION );
+	if ( isset( $metadata['file'] ) ) {
+		$filename = pathinfo( $metadata['file'], PATHINFO_FILENAME );
+	} else {
+		$filename = pathinfo( $file, PATHINFO_FILENAME );
+	}
```

Comment:
Sure, this condition simplifies a section of code that was used before to create the `filename`. The condition ensures that the `file` exists otherwise it fallbacks to the full `$file` path.

This was before:

```diff
---	$original_extension = pathinfo( $file, PATHINFO_EXTENSION );
---	$filename         = wp_basename( $file, ".{$original_extension}" );
+++	if ( isset( $metadata['file'] ) ) {
+++		$filename = pathinfo( $metadata['file'], PATHINFO_FILENAME );
+++	} else {
+++		$filename = pathinfo( $file, PATHINFO_FILENAME );
+++	}
 
```

However, both lines can be simplified by just using `PATHINFO_FILENAME` from the `file` if present otherwise tries with `$file`

Reply:
@mitogh Hmm, the previous version was not using an `if` conditional, so it was actually simpler than how you changed it now.

My original question though was more related to why did you even add `$metadata['file']` here? The code before didn't pay attention to `$metadata['file']` and instead only relied on the attached file `$file`, so why did you add this? The question is less about the code complexity, but more about what the reasoning was to even consider `$metadata['file']` when it previously relied on `$file` only. Hope that makes sense.

Reply:
Another related question: Why do you use this condition to set the `$filename` but not the `$original_directory`? It seems inconsistent to use one file for one thing and potentially another file for the other.

Reply:
> Hmm, the previous version was not using an if conditional, so it was actually simpler than how you changed it now.

Correct due now I'm accessing `file` property I first make sure this property exists. 

> My original question though was more related to why did you even add $metadata['file'] here? The code before didn't pay attention to $metadata['file'] and instead only relied on the attached file $file, so why did you add this? T

The problem was due `file` was using the original file, so `$file` and 'file` property where not quite the same, looking into this it does not makes sense to keep using the property as `$file` would be always the attached file instead.

Reply:
> Another related question: Why do you use this condition to set the $filename but not the $original_directory? It seems inconsistent to use one file for one thing and potentially another file for the other.

Mostly because the directory can be determined by just using the file, after a closer look at this, the conditional is not required and we can keep on using the `$file` variable instead, thanks for flagging this one.